File: archivist/src/test/java/com/zorroa/archivist/SecurityUtilsTest.java
package com.zorroa.archivist;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.Lists;
import com.zorroa.archivist.domain.Access;
import com.zorroa.archivist.domain.Acl;
import com.zorroa.archivist.domain.Permission;
import com.zorroa.archivist.sdk.security.Groups;
import com.zorroa.archivist.security.UtilsKt;
import com.zorroa.archivist.service.UserService;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class SecurityUtilsTest extends AbstractTest {
    @Autowired
    UserService userService;

    @Test
    public void testHasPermissionWithAcl() {
        // Get the perms we have
        authenticate("user");
        List<Integer> perms = ImmutableList.copyOf(UtilsKt.getPermissionIds());

        // Create an ACL and add a Read permission
        Acl acl = new Acl();
        acl.addEntry(perms.get(0), Access.Read);

        // Test the has Permission function.
        assertTrue(UtilsKt.hasPermission(acl, Access.Read));
        assertFalse(UtilsKt.hasPermission(acl, Access.Write));
    }

    @Test
    public void testHasPermissionIsAdministrator() {
        Permission p = permissionService.getPermission(Groups.ADMIN);
        userService.setPermissions(UtilsKt.getUser(), Lists.newArrayList(p));

        // Reauthenticate the user, this sets up the admin's normal
        // permissions.
        authenticate("admin");

        // Create an ACL and add a Read permission
        List<Integer> perms = ImmutableList.copyOf(UtilsKt.getPermissionIds());
        Acl acl = new Acl();
        acl.addEntry(perms.get(0), Access.Read);

        // Test the hasPermission function.
        assertTrue(UtilsKt.hasPermission(acl, Access.Read));
        assertTrue(UtilsKt.hasPermission(acl, Access.Write));
    }
}

File: archivist/src/test/kotlin/com/zorroa/archivist/SecurityUtilsTest.kt
package com.zorroa.archivist

import com.google.common.collect.ImmutableList
import com.google.common.collect.Lists
import com.zorroa.archivist.domain.Access
import com.zorroa.archivist.domain.Acl
import com.zorroa.archivist.domain.Permission
import com.zorroa.security.Groups
import com.zorroa.archivist.security.*
import com.zorroa.archivist.service.UserService

import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import java.util.UUID

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue

class SecurityUtilsTest : AbstractTest() {

    @Test
    fun testHasPermissionWithAcl() {
        // Get the perms we have
        authenticate("user@zorroa.com")
        val perms = ImmutableList.copyOf(getPermissionIds())

        // Create an ACL and add a Read permission
        val acl = Acl()
        acl.addEntry(perms[0], Access.Read)

        // Test the has Permission function.
        assertTrue(hasPermission(acl, Access.Read))
        assertFalse(hasPermission(acl, Access.Write))
    }

    @Test
    fun testHasPermissionIsAdministrator() {
        val p = permissionService.getPermission(Groups.ADMIN)
        userService.setPermissions(getUser(), Lists.newArrayList(p))

        // Reauthenticate the user, this sets up the admin's normal
        // permissions.
        authenticate("admin")

        // Create an ACL and add a Read permission
        val perms = ImmutableList.copyOf(getPermissionIds())
        val acl = Acl()
        acl.addEntry(perms[0], Access.Read)

        // Test the hasPermission function.
        assertTrue(hasPermission(acl, Access.Read))
        assertTrue(hasPermission(acl, Access.Write))
    }
}
